server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java

```diff
@@ -221,23 +221,11 @@ public void testCancelMultiSearch() throws Exception {
         }
     }
 
-    /**
-     * The test `testCancelFailedSearchWhenPartialResultDisallowed` usually fails when concurrency is enabled unless
-     * the `cancelledLatch.await()` section is commented out. However, this approach seems prone to race conditions.
-     * Further investigation is needed to determine if this test just needs to be revised, or rather, if it is
-     * detecting a deeper issue.  For now, we will disable concurrency here.
-     */
-    @Override
-    protected boolean enableConcurrentSearch() {
-        return false;
-    }
-
     public void testCancelFailedSearchWhenPartialResultDisallowed() throws Exception {
         final List<ScriptedBlockPlugin> plugins = initBlockFactory();
         int numberOfShards = between(2, 5);
         AtomicBoolean failed = new AtomicBoolean();
         CountDownLatch queryLatch = new CountDownLatch(1);
-        CountDownLatch cancelledLatch = new CountDownLatch(1);
```

Comment:
The threading model has changed, and if my understanding is correct, this now blocks the shard worker threads from returning, as the check for cancellation happens prior to this block being run.